rbac/ledger_sync/deltas/handlers.py

```diff
@@ -48,21 +47,21 @@ def update_database(database, state_change):
                 resource = data_to_dicts(change.address, change.value)[0]
                 data_type = addresser.get_address_type(change.address)
                 if data_type in TABLE_NAMES and TABLE_NAMES[data_type] == "roles":
-                    database.clear_role(resource["role_id"], resource["created_date"])
+                    clear_role(conn, resource["role_id"], resource["created_date"])
                 update(change.address, resource)
 
 
-def _handle_state_changes(database, state_change):
+def _handle_state_changes(conn, state_change):
     """Takes in a database object and sawtooth state change, parses changes, and
     updates any changed objects in rethinkdb.
     """
     try:
         # Check for and resolve forks
         state_change.block_num = int(state_change.block_num)
-        old_block = database.fetch("blocks", state_change.block_num)
+        old_block = r.table("blocks").get(state_change.block_num).run(conn)
         if old_block is not None:
             if old_block["block_id"] != state_change.block_id:
-                drop_results = database.drop_fork(state_change.block_num)
+                drop_results = conn.drop_fork(state_change.block_num)
```

Comment:
The connection doesn't have a drop_fork() function. You would need to call the drop_fork() function that you created in line 98 of this file.